rasa/shared/data.py

```diff
@@ -45,3 +48,210 @@ def is_likely_markdown_file(file_path: Text) -> bool:
+            data_files.update(new_data_files)
+
+    return sorted(data_files)
+
+
+def _find_data_files_in_directory(
+    directory: Text, filter_property: Callable[[Text], bool]
+) -> Set[Text]:
+    filtered_files = set()
+
+    for root, _, files in os.walk(directory, followlinks=True):
+        # we sort the files here to ensure consistent order for repeatable training
+        # results
+        for f in sorted(files):
+            full_path = os.path.join(root, f)
+
+            if not is_valid_filetype(full_path):
+                continue
+
+            if filter_property(full_path):
+                filtered_files.add(full_path)
+
+    return filtered_files
+
+
+def is_valid_filetype(path: Union[Path, Text]) -> bool:
+    """Checks if given file has a supported extension.
+    Args:
+        path: Path to the source file.
+    Returns:
+        `True` is given file has supported extension, `False` otherwise.
+    """
+    return Path(path).is_file() and Path(path).suffix in TRAINING_DATA_EXTENSIONS
+
+
+def is_nlu_file(file_path: Text) -> bool:
+    """Checks if a file is a Rasa compatible nlu file.
+
+    Args:
+        file_path: Path of the file which should be checked.
+
+    Returns:
+        `True` if it's a nlu file, otherwise `False`.
+    """
+    from rasa.shared.nlu.training_data import loading as nlu_loading
+
+    return nlu_loading.guess_format(file_path) != nlu_loading.UNK
+
+
+def is_story_file(file_path: Text) -> bool:
+    """Checks if a file is a Rasa story file.
+
+    Args:
+        file_path: Path of the file which should be checked.
+
+    Returns:
+        `True` if it's a story file, otherwise `False`.
+    """
+    from rasa.shared.core.training_data.story_reader.yaml_story_reader import (
+        YAMLStoryReader,
+    )
+    from rasa.shared.core.training_data.story_reader.markdown_story_reader import (
+        MarkdownStoryReader,
+    )
+
+    return YAMLStoryReader.is_yaml_story_file(
+        file_path
+    ) or MarkdownStoryReader.is_markdown_story_file(file_path)
+
+
+def is_test_stories_file(file_path: Text) -> bool:
+    """Checks if a file is a test stories file.
+
+    Args:
+        file_path: Path of the file which should be checked.
+
+    Returns:
+        `True` if it's a story file containing tests, otherwise `False`.
+    """
+    from rasa.shared.core.training_data.story_reader.yaml_story_reader import (
+        YAMLStoryReader,
+    )
+    from rasa.shared.core.training_data.story_reader.markdown_story_reader import (
+        MarkdownStoryReader,
+    )
+
+    return YAMLStoryReader.is_yaml_story_file(
+        file_path
+    ) or MarkdownStoryReader.is_markdown_test_stories_file(file_path)
+
+
+def is_config_file(file_path: Text) -> bool:
+    """Checks whether the given file path is a Rasa config file.
+
+       Args:
+           file_path: Path of the file which should be checked.
+
+       Returns:
+           `True` if it's a Rasa config file, otherwise `False`.
+       """
```

Comment:
```suggestion
    Args:
        file_path: Path of the file which should be checked.

    Returns:
        `True` if it's a Rasa config file, otherwise `False`.
    """
```